smartconfig/file.py

```diff
@@ -65,7 +65,7 @@ def load(path: _FilePath) -> None:
 
     for root_node_name, root_node_value in yaml_content.items():
         if not isinstance(root_node_value, dict):
-            raise ...
+            raise ConfigurationError(f"Cannot set the attribute {root_node_name} on the root node.")
```

Comment:
On second thought, it's better to ignore non-mapping nodes under the root; nodes elsewhere may hold references to such nodes. In any case their presence isn't an inconvenience to the parser.